assignment/assignment.hs

```diff
@@ -26,8 +26,12 @@ import Data.Char(intToDigit)
 -- predicate returns `True`. Implement `myfilter` using `foldl`. (Hint: you 
 -- might need to use `++` rather than `:` to build the result)
 
--- myfilter :: (a -> Bool) -> [a] -> [a]
 
+select pred a x = 
+  if pred x
+    then a ++ [x] --looks like I did
+    else a
+myfilter pred a = foldl (select pred) [] a
 
```

Comment:
Yup.

Stylistically you could put `select` into a where clause, which would localize it to this one use.